addons/website/static/src/interactions/cookies/cookies_bar.js

```diff
@@ -0,0 +1,123 @@
+import { cookie } from "@web/core/browser/cookie";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { isVisible } from "@web/core/utils/ui";
+import wUtils from "@website/js/utils";
+import { Popup } from "../popup/popup";
+// import { setUtmsHtmlDataset } from "@website/js/content/inject_dom";
+
+// Extending the Popup class with cookiebar functionality.
+// This allows for refusing optional cookies for now and can be
+// extended to picking which cookies categories are accepted.
+export class CookiesBar extends Popup {
+    static selector = "#website_cookies_bar";
+    dynamicContent = Object.assign(this.dynamicContent, {
+        "#cookies-consent-essential, #cookies-consent-all": {
+            "t-on-click": this.onAcceptClick,
+        },
+        // Override to avoid side effects on hide.
+        ".js_close_popup": {
+            "t-on-click": () => {},
+        },
+    });
+
+    setup() {
+        super.setup();
+        this.showToggle();
+    }
+
+    start() {
+        super.start();
+        this.addListener(
+            this.services.website_cookies.bus,
+            "cookiesBar.show",
+            this.onShowCookiesBar
+        );
+        this.addListener(
+            this.services.website_cookies.bus,
+            "cookiesBar.toggle",
+            this.toggleCookiesBar,
+        );
+    }
+
+    showPopup() {
+        super.showPopup();
+        if (this.toggleEl) {
+            this.toggleCookiesBar();
+        }
+    }
+
+    showToggle() {
+        const policyLinkEl = this.el.querySelector(".o_cookies_bar_text_policy");
+        if (policyLinkEl && window.location.pathname === new URL(policyLinkEl.href).pathname) {
+            this.toggleEl = wUtils.cloneContentEls(`
+            <button class="o_cookies_bar_toggle btn btn-info btn-sm rounded-circle d-flex gap-2 align-items-center position-fixed pe-auto">
+                <i class="fa fa-eye" alt="" aria-hidden="true"></i> <span class="o_cookies_bar_toggle_label"></span>
+            </button>
+            `).firstElementChild;
+            this.insert(this.toggleEl, this.el, "beforebegin");
+            this.services.website_core.startInteractions(this.toggleEl);
+        }
+    }
+
+    toggleCookiesBar() {
+        this.bsModal.toggle();
+        // As we're using Bootstrap's events, the Popup class prevents the modal
+        // from being shown after hiding it: override that behavior.
+        this._popupAlreadyShown = false;
+        cookie.delete(this.el.id);
+    }
+
+    /**
+     * @param {Event} ev
+     */
+    onAcceptClick(ev) {
+        const isFullConsent = ev.target.id === "cookies-consent-all";
+        this.cookieValue = `{"required": true, "optional": ${isFullConsent}}`;
+        if (isFullConsent) {
+            document.dispatchEvent(new Event("optionalCookiesAccepted"));
```

Comment:
not sure, but would it not be better to use the bus?
```js
this.services.website_cookies.bus.trigger("optionalCookiesAccepted");
```

Reply:
Not on this one I think, because it really has to be dispatched on the document for other uses outside of interactions. The idea is also to allow users to add custom js code that would listen to this event. (even though it's not yet documented right now...)